### PR TITLE
Fix/improvetestcoverage

### PR DESCRIPTION
**Pull request by wroel**
Improved test coverage as much as possible, deleted redundant if-statements and inline comments.

### CODE REVIEW

File: src/main/java/nl/tudelft/ti2206/semgr8/model/level/LevelMechanics.java [hunk 1]
```diff
@@ -79,15 +77,11 @@ public boolean decideIfStone() {
...
-            return Gem.START_AMOUNT_OF_COLORS;
-        }
+        int max = Gem.MAX_AMOUNT_OF_COLORS;
+        double l = (double)level.getLevel();
+        double increase = (double)INCREASE_AMOUNT_OF_COLORS;
+        double change = (double)LEVEL_CHANGE_AT_LEVELS;
+        int current = Gem.START_AMOUNT_OF_COLORS + (int)Math.floor((l + increase) / change);
+        return Math.min(max, current);
     }
 }
```
Comment (albert-smit) [inline src/main/java/nl/tudelft/ti2206/semgr8/model/level/LevelMechanics.java]: missing newline

File: src/test/java/nl/tudelft/ti2206/semgr8/model/board/BoardTest.java [hunk 2]
```diff
@@ -106,6 +120,28 @@ public void testBoardDimensions() {
         assertEquals(board.getGridDimensions(), 8);
     }
 
+    /**
+     * Test if decreaseGridDimension decreases the dimensions of Board when allowed.
+     */
+    @Test
+    public void testDecreaseGridDimension1() {
```
Comment (albert-smit) [inline src/test/java/nl/tudelft/ti2206/semgr8/model/board/BoardTest.java]: these three tests could get better names

File: src/test/java/nl/tudelft/ti2206/semgr8/model/piece/LaserGemTest.java [hunk 3]
```diff
@@ -39,4 +42,19 @@ public void testToString() {
...
+        b.setPiece(0, 2, new LaserGem(0, 2, 0));
+        LaserGem laserGem = (LaserGem)b.getGem(0, 2);
+        laserGem.doPowerUp(b);
+        for (int i = 0; i < b.getGridDimensions(); i++) {
+            assertTrue(b.getPiece(i, 2).isEmpty());
+            assertTrue(b.getPiece(0, i).isEmpty());
+        }
+        assertEquals(350, b.getScore());
+    }
 }
```
Comment (albert-smit) [inline src/test/java/nl/tudelft/ti2206/semgr8/model/piece/LaserGemTest.java]: missing newline

File: src/main/java/nl/tudelft/ti2206/semgr8/model/level/LevelMechanics.java [hunk 4]
```diff
@@ -79,15 +77,11 @@ public boolean decideIfStone() {
...
-        } else if (level.getLevel() >= NINE_COLORS_AT_LEVEL) {
-            return Gem.MAX_AMOUNT_OF_COLORS;
-        } else {
-            return Gem.START_AMOUNT_OF_COLORS;
-        }
+        int max = Gem.MAX_AMOUNT_OF_COLORS;
+        double l = (double) level.getLevel();
+        double increase = (double) INCREASE_AMOUNT_OF_COLORS;
+        double change = (double) LEVEL_CHANGE_AT_LEVELS;
+        int current = Gem.START_AMOUNT_OF_COLORS + (int) Math.floor((l + increase) / change);
```
Comment (kiliangrashoff) [inline src/main/java/nl/tudelft/ti2206/semgr8/model/level/LevelMechanics.java]: Use more logical constants, with easier names, like START_LEVEL_COLORS_INCREASE and INTERVAL_COLORS_INCREASE.

**Comment by kiliangrashoff** [comment]
Looks good besides the comments

**Comment by kiliangrashoff** [comment]
Fixed some last errors, mergin after build completes.

**Pull request merged**